hello-world/src/worker.ts

```diff
@@ -1,22 +1,31 @@
 // @@@SNIPSTART typescript-hello-worker
-import { Worker } from '@temporalio/worker';
+import { NativeConnection, Worker } from '@temporalio/worker';
 import * as activities from './activities';
 
 async function run() {
-  // Step 1: Register Workflows and Activities with the Worker and connect to
-  // the Temporal server.
+  // Step 1: Establish a connection with Temporal server.
```

Comment:
```suggestion
  // Step 1: Establish a connection with Temporal server.
  //
  // NativeConnection is implemented in Rust and is used by Core SDK which powers the Worker.
```

Reply:
Is there any benefit in mentioning Rust and Core SDK here? It's strictly implementation detail, so if these changes are aimed at helping less expert users it may be best not to. E.g. it might trigger thoughts like "Do I need Rust?", "Where do I get Core SDK?"

Reply:
Agreed it might be pertinent to make this text just a little bit easier to understand for new users, but we somehow need to make it clear that `Connection` is for client calls, and `NativeConnection` is for Workers. And unfortunately, the `Native` prefix here may sounds a bit mysterious and isn't helping remembering that difference, unless the user understands that the Worker relies on some native code (aka Core SDK, coded in Rust).

Reply:
I see. I'd suggest something like

```
Worker code uses `NativeConnection`. 
(But in your application code it's just `Connection`.)
```

Reply:
I like this

Reply:
I really hope we can deprecate this class soon when Core can delegate gRPC calls to lang.